app/controllers/shipments_controller.rb

```diff
@@ -0,0 +1,72 @@
+class ShipmentsController < ApplicationController
+
+  def shipment
+    submission_errors = submission_error_handling(params)
+
+    if submission_errors
+      return submission_errors
+    else
+      origin = ActiveShipping::Location.new(country: params[:origin][:country], state: params[:origin][:state], city: params[:origin][:city], zip: params[:origin][:zip])
+
+      destination = ActiveShipping::Location.new(country: params[:destination][:country], state: params[:destination][:state], city: params[:destination][:city], zip: params[:destination][:zip])
+
+      packages = []
+      if params[:packages].length > 0
+        params[:packages].each do |package|
+          weight = package[:weight].to_i
+          dimensions = package[:dimensions].split(",").map!{|string| string.to_i}
+          pack = ActiveShipping::Package.new(weight, dimensions)
+          packages << pack
+        end
+          @quotes = get_quotes(origin, destination, packages)
+          render :json => @quotes.as_json
```

Comment:
These two lines are outside the `params[:packages].each` block, but they are indented like they are inside the block. Spacing can make a huge difference in readability for code, so I recommend adding an extra line after a block and being consistent with indentation.